demo/cmd/retry/retry.go

```diff
@@ -0,0 +1,40 @@
+package main
```

Comment:
Why do we need a custom app to test retries?

Shouldn't this be possible with `httpbin` and the `curl` client where we can simulate failures by curling `/status/<status-code>`?